src/state/slices/stakingDataSlice/stakingDataSlice.test.ts

```diff
@@ -0,0 +1,522 @@
+      })
+    })
+
+    describe('selectNonloadedValidators', () => {
+      it('returns empty array when state is empty', async () => {
+        const selected = selectNonloadedValidators(store.getState(), cosmosAccountSpecifier)
+        expect(selected).toEqual([])
+      })
+
+      it('returns all validators in staking data when staking data is loaded but validators are not', async () => {
+        store.dispatch(
+          stakingData.actions.upsertStakingData({
+            accountSpecifier: cosmosAccountSpecifier,
+            stakingData: mockStakingData
+          })
+        )
+
+        const selected = selectNonloadedValidators(store.getState(), cosmosAccountSpecifier)
+        expect(selected).toEqual([
+          'cosmosvaloper1qtxec3ggeuwnca9mmngw7vf6ctw54cppey02fs',
+          'cosmosvaloper199mlc7fr6ll5t54w7tts7f4s0cvnqgc59nmuxf',
+          'cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn'
+        ])
+      })
+
+      describe('validators and staking data are loaded', () => {
+        beforeEach(() => {
+          store.dispatch(
+            stakingData.actions.upsertValidatorData({
+              validators: mockValidatorData
+            })
+          )
+        })
+
+        it('returns empty array when all validators in stakingData are loaded / known', async () => {
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+
+          const selected = selectNonloadedValidators(store.getState(), cosmosAccountSpecifier)
+          expect(selected).toEqual([])
+        })
+
+        it('returns correct array when some validators in stakingData are not loaded / unknown', async () => {
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingWithUnknownValidators
+            })
+          )
+
+          const selected = selectNonloadedValidators(store.getState(), cosmosAccountSpecifier)
+          expect(selected).toEqual([
+            'cosmosvaloper1r9lxkpqre6j4487ut882xchgr7rdtx3x76gtdp',
+            'cosmosvaloper1yvwqd5rdtuaw25mcqhz794dvgq9k9yeh8mjcdh'
+          ])
+        })
+      })
+    })
+
+    describe('selectSingleValidator', () => {
+      it('returns null when state is empty', async () => {
+        const selected = selectSingleValidator(
+          store.getState(),
+          cosmosAccountSpecifier,
+          SHAPESHIFT_VALIDATOR_ADDRESS
+        )
+        expect(selected).toBeNull()
+      })
+
+      describe('validators and staking data are loaded', () => {
+        beforeEach(() => {
+          store.dispatch(
+            stakingData.actions.upsertValidatorData({
+              validators: mockValidatorData
+            })
+          )
+        })
+
+        it('returns null when validator address is unknown', async () => {
```

Comment:
What does unknown mean from a code perspective? Validator not present in the slice? From Cosmos/unchained/chain-adapters perspective, it is not unknown
<img width="1233" alt="image" src="https://user-images.githubusercontent.com/17035424/162176080-f7aad212-9ae7-400b-b752-b895fe318773.png">

Reply:
```suggestion
        it('returns null when validator data is not present in state', async () => {
```